src/main/kotlin/org/rust/lang/core/completion/RsCommonCompletionProvider.kt

```diff
@@ -411,3 +415,37 @@ private class RsImportLookupElement(
         return result
     }
 }
+
+fun collectVariantsForEnumCompletion(
+    element: RsEnumItem, context: RsCompletionContext, substitution: Substitution, candidate: ImportCandidate? = null
+): List<LookupElement> {
+    val enumName = element.name ?: return emptyList()
+
+    return element.enumBody?.childrenOfType<RsEnumVariant>().orEmpty().mapNotNull { enumVariant ->
+        val variantName = enumVariant.name ?: return@mapNotNull null
+
+        return@mapNotNull createLookupElement(
+            scopeEntry = SimpleScopeEntry("${enumName}::${variantName}", enumVariant, substitution),
+            context = context,
+            null,
+            object : RsDefaultInsertHandler() {
+                override fun handleInsert(element: RsElement, scopeName: String, context: InsertionContext, item: LookupElement) {
+                    // move start offset from enum to its variant to escape it
+                    val enumStartOffset = context.startOffset
+                    val variantStartOffset = enumStartOffset + (enumName.length + 2)
+                    context.offsetMap.addOffset(CompletionInitializationContext.START_OFFSET, variantStartOffset)
+                    super.handleInsert(enumVariant, variantName, context, item)
+
+                    // escape enum name if needed
+                    if (element is RsNameIdentifierOwner && !RsNamesValidator.isIdentifier(enumName) && enumName !in CAN_NOT_BE_ESCAPED)
```

Comment:
Please add some test(s) for this route.

Reply:
done. I added couple of tests for this method